src/Drips.sol

```diff
@@ -666,22 +694,22 @@ library Drips {
     /// @param amtPerSec The dripping rate
     function _addDelta(
         mapping(uint32 => AmtDelta) storage amtDeltas,
-        uint32 cycleSecs,
-        uint32 timestamp,
+        uint256 cycleSecs,
+        uint256 timestamp,
         int256 amtPerSec
     ) private {
         unchecked {
-            AmtDelta storage amtDelta = amtDeltas[_cycleOf(timestamp, cycleSecs)];
+            AmtDelta storage amtDelta = amtDeltas[_cycleOf(uint32(timestamp), uint32(cycleSecs))];
             int256 thisCycleDelta = amtDelta.thisCycle;
             int256 nextCycleDelta = amtDelta.nextCycle;
 
             // In order to set a delta on a specific timestamp it must be introduced in two cycles.
             // The cycle delta is split proportionally based on how much this cycle is affected.
             // The next cycle has the rest of the delta applied, so the update is fully completed.
-            uint32 nextCycleSecs = timestamp % cycleSecs;
-            uint32 thisCycleSecs = cycleSecs - nextCycleSecs;
-            thisCycleDelta += int256(uint256(thisCycleSecs)) * amtPerSec;
-            nextCycleDelta += int256(uint256(nextCycleSecs)) * amtPerSec;
+            int256 amtPerCycle = (amtPerSec * int256(cycleSecs)) / (1 << 64);
+            int256 amtSplitCycle = (amtPerSec * int256(timestamp % cycleSecs)) / (1 << 64);
```

Comment:
The only bit shifting related to fractions happens here but would not the compiler do the work here once and replace it with a constant?